src/characters/enrico.ts

```diff
@@ -1,20 +1,29 @@
-import { MessageKit, PublicKey, SecretKey } from '@nucypher/nucypher-core';
+import { PublicKey, SecretKey } from '@nucypher/nucypher-core';
 
+import { MessageKit } from '../core';
+import { ConditionSet } from '../policies/conditions';
 import { toBytes } from '../utils';
 
 export class Enrico {
   public readonly policyEncryptingKey: PublicKey;
   public readonly verifyingKey: PublicKey;
+  public readonly conditions?: ConditionSet;
 
-  constructor(policyEncryptingKey: PublicKey, verifyingKey?: PublicKey) {
+  constructor(
+    policyEncryptingKey: PublicKey,
+    verifyingKey?: PublicKey,
+    conditions?: ConditionSet
+  ) {
     this.policyEncryptingKey = policyEncryptingKey;
     this.verifyingKey = verifyingKey ?? SecretKey.random().publicKey();
+    if (conditions) this.conditions = conditions;
   }
 
   public encryptMessage(plaintext: Uint8Array | string): MessageKit {
     return new MessageKit(
       this.policyEncryptingKey,
-      plaintext instanceof Uint8Array ? plaintext : toBytes(plaintext)
+      plaintext instanceof Uint8Array ? plaintext : toBytes(plaintext),
+      this.conditions
```

Comment:
I like that you can just set conditions for Enrico when he's created, and then forget about them for all of the encryptions.

could we also include the ability to pass in a `ConditionsSet` to `encryptMessage` to make it easier to have different conditions on each encryption?

Reply:
Yes I can't see a downside there, certainly does add some possibilities.

Reply:
From an API perspective, perhaps a little nuance here...if Enrico takes a ConditionSet on initialization, from an API perspective what does the ConditionSet passed into `encryptMessage` mean? Does it mean an override of the original ConditionSet i.e. use only the ConditionSet provided as the method parameter and not the one from initialization for this message? OR a ConditionSet that is added onto the ConditionSet used for initialization for encrypting this specific message i.e. aggregation of Conditions for encrypting this message...?

In the override case, the API would seem a little obscure to me - temporarily overriding an instance variable. For such a use case are you envisioning it being too bulky or unreasonable to just initialize a new temporary Enrico for encryption that uses a different condition?

Reply:
Yes I agree @derekpierre,  creating Enricos per conditions seemed lightweight enough to me that it made sense to do it this way for now at least.